### DISCUSSION TITLE
sempy_labs.backup_semantic_model

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/semantic-link-labs#356] [General]
I'm trying to run this: sempy_labs.backup_semantic_model

But I don't see a notebook. Can anyone help me get the Python?

**Comment by m-kovalsky** [comment]
First, make sure you have an ADLS Gen2 storage account connected to your workspace. Then use this code as a template.
```python
import sempy_labs as labs
dataset = '' # Enter the name of your semantic model
file_path = '' # Enter the file path you want to use for backing up the model. i.e.: 'MyModel.abf' or 'MyFolder/MyModel.abf'
workspace = '' # Enter the name of the workspace in which the semantic model exists
labs.backup_semantic_model(dataset=dataset, file_path=file_path, workpsace=workspace)
```

**Reply by ghost** [reply]
Thanks @m-kovalsky, my company has asked me to do everything using service principal accounts so that it works even if my account changes. Can you help me understand how to do this using a service principal account. Also, is there a way to see the python behind this package?

**Comment by m-kovalsky** [comment]
Service principal support is coming - it has been highly requested (#171). All the code for this library is [open source](https://github.com/microsoft/semantic-link-labs/tree/main/src/sempy_labs).

**Comment by ghost** [comment]
@m-kovalsky  I see that there is also the option to restore a dataset? The documentation shows that it only restores as the same dataset name. Is there a way to take an ".abf" file and save it as a new semantic model?

[dataset=dataset, file_path = file_path, allow_overwrite = True, workspace = workspace](https://semantic-link-labs.readthedocs.io/en/stable/sempy_labs.html#sempy_labs.restore_semantic_model)

**Comment by m-kovalsky** [comment]
In that function, the 'file_path' parameter is the name of the .abf (backup file) to restore. The 'dataset' parameter will be the name of the restored dataset.

**Reply by ghost** [reply]
@m-kovalsky that's what I thought should be the case but when I apply your suggestion I get an error. Here is my code below. when I try to back up a dataset THAT EXISTS with a file it works. BUT when I try to create a new dataset with the same file it gives me this error **"ValueError: There's no item with the name 'test SalesForce SFDC' in workspace 'HCH_DEV Dataset COMPASS'"**

**CODE THAT WORKS: This dataset already exists and I am just restoring it from a file.**
dataset = 'SalesForce SFDC' # Enter the name of your semantic model
file_path = 'Jan 03, 2025_SalesForce SFDC-HCH_DEV Dataset COMPASS.abf' # Enter the file path you want to use for backing up the model. i.e.: 'MyModel.abf' or 'MyFolder/MyModel.abf'
workspace = 'HCH_DEV Dataset COMPASS' # Enter the name of the workspace in which the semantic model exists
labs.restore_semantic_model(dataset = dataset, file_path = file_path, allow_overwrite = True, ignore_incompatibilities = True, force_restore = False, workspace = workspace)

**CODE THAT DOES NOT WORK: Notice the only change is the dataset name:**
dataset = 'test SalesForce SFDC' # Enter the name of your semantic model
file_path = 'Jan 03, 2025_SalesForce SFDC-HCH_DEV Dataset COMPASS.abf' # Enter the file path you want to use for backing up the model. i.e.: 'MyModel.abf' or 'MyFolder/MyModel.abf'
workspace = 'HCH_DEV Dataset COMPASS' # Enter the name of the workspace in which the semantic model exists
labs.restore_semantic_model(dataset = dataset, file_path = file_path, allow_overwrite = True, ignore_incompatibilities = True, force_restore = False, workspace = workspace)
